uwblueprint/childrens-aid-society
# Squash Goals and Concerns Tables

## Notion ticket link
[Condense goals and concerns tables](https://www.notion.so/uwblueprintexecs/Condense-goals-and-concerns-tables-ded314841c9f49f28241e9b686343ed7)


## Implementation description
* Squash familial concerns and child concerns models into one concerns model. Add a concerns_type enum to differentiate between different concerns in the unified concerns model
* Set up concerns_intake table with intake_id and concern_id foreign keys
* Squash long term and short terms goal models into a unified goals model. Add a goals_type enum to differentiate between different goals in the goals model.
* Modify alembic migration file to ensure data is seeded correctly upon db upgrade and db downgrade



## Steps to test
1. Restart your docker containers: docker-compose down --volumes && docker-compose up
2. Exec into the backend container: docker exec -it cas_py_backend /bin/bash
3. Upgrade the database: flask db upgrade -- verify that it works successfully
4. Downgrade and upgrade the database again: flask db downgrade && flask db upgrade -- verify that it works successfully
5. Exec into the db container: docker exec -it cas_db psql -U postgres -d cas
6. Verify that `concerns`, `goals`, and `intakes_concerns` tables are present as shown below. Also verify that old `child_concerns`, `familial_concerns`, `long_term_goals`, and `short_term_goals` tables are not present.
<img width="432" alt="image" src="https://user-images.githubusercontent.com/55464415/177668304-ee575fd1-5ce3-4d2b-be00-576d634f650a.png">
7. Downgrade the database: flask db downgrade. Verify `child_concerns`, `familial_concerns`, `long_term_goals`, and `short_term_goals` tables have been restored.



## What should reviewers focus on?
* concerns_intake table has the correct foreign keys set
* Tables have the correct columns in postgres


## Checklist
